tests/test_steps_edge.py

```
@@ -109,3 +126,150 @@ def test_render_table_view_nan_std_direct():
+        with patch("pandas.read_csv", return_value=pd.DataFrame({"Wrong": [1]})):
+            success, sig = steps._process_uploaded_file(mock_file)
+            assert not success
+            mock_st.error.assert_called_with(
+                "File missing required columns: Name and Score*"
+            )
+
+
+def test_render_step_1_empty_warning():
+    """Verify warning when attempting to proceed from Step 1 with empty data."""
+    with patch("src.ui.steps.st") as mock_st:
+        mock_st.button.side_effect = lambda label, **kwargs: label == "Next: Configure"
```

Ruff (0.15.12)

[warning] Unused lambda argument: `kwargs`

(ARG005)